lib/nrf_modem_lib/nrf_modem_os.c

```
@@ -277,6 +277,49 @@ unsigned int nrf_modem_os_sem_count_get(void *sem)
 	return k_sem_count_get(sem);
 }
 
+static struct k_mutex nrf_modem_os_mutexes[NRF_MODEM_OS_NUM_MUTEX_REQUIRED];
+
+int nrf_modem_os_mutex_init(void **mutex)
+{
+	static uint8_t used;
+
+	if (PART_OF_ARRAY(nrf_modem_os_mutexes, (struct k_mutex *)*mutex)) {
+		goto recycle;
+	}
+
+	__ASSERT(used < NRF_MODEM_OS_NUM_MUTEX_REQUIRED,
+		 "Not enough mutexes in glue layer");
+
+	*mutex = &nrf_modem_os_mutexes[used++];
+
+recycle:
+	return k_mutex_init((struct k_mutex *)*mutex);
+}
+
+void nrf_modem_os_mutex_unlock(void *mutex)
+{
+	__ASSERT(PART_OF_ARRAY(nrf_modem_os_mutexes, (struct k_mutex *)mutex),
+		 "Uninitialised mutex");
+
+	k_mutex_unlock((struct k_mutex *)mutex);
```

`k_mutex_unlock` can return errors. I believe this should propagate any error to the user.